woocommerce/woocommerce-android
# Activate: Store creation done local notification

Fixes #9060, a subtask of #8999.

This PR integrates the WP.com feature flag repository and fetches the flags on app initialization. `LOCAL_NOTIFICATION_STORE_CREATION_READY` is then used to check if a local notification should be scheduled 5 minutes after the store creation is started.

The `WorkManager` worker chain has been updated and it now contains 2 steps:

1. `PreconditionWorker` - This worker is delayed by the configured amount of time, upon which a precondition check is performed. If the precondition is satisfied, the final step is executed, otherwise the work is canceled.
3. `LocalNotificationWorker` - This is the point when a local notification is displayed.

**To test:**
1. Start the store creation flow (either from the prologue or site picker)
2. Complete all the steps and sign up for a free trial
3. Put the app in the background and wait for > 5 minutes
4. Notice a notification is displayed
5. Tap on the notification and the new store is loaded
6. Repeat the store creation flow but now keep the app running in the foreground
7. Wait for > 5 minutes
8. Notice the new store is automatically loaded and the notification is not displayed

> **Note**
> Please only merge after the target branch is set to `trunk`
